app/controllers/users_controller.rb

```diff
@@ -2,12 +2,16 @@
 class UsersController < ApplicationController
   load_permissions_and_authorize_resource
   before_action :set_user
+
   def index
   end
 
   def show
   end
 
+  def profile
```

Comment:
You should do that with the help of cancan.

`show_the_profile_information`
`show_edit_fields if can? :manage, @user`